build.gradle.kts

```diff
@@ -0,0 +1,232 @@
+}
+
+apply(plugin = "maven-publish")
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+dependencies {
+    compileOnly("org.scoverage:scalac-scoverage-plugin_2.13:1.4.2")
+    implementation(group = "commons-io", name = "commons-io", version = "2.6")
+
+    implementation(kotlin("script-runtime"))
+    testImplementation(kotlin("test"))
+
+    testImplementation("junit:junit:4.12")
+    testImplementation("org.junit.jupiter:junit-jupiter-api:5.5.2")
+    testRuntimeOnly("org.junit.vintage:junit-vintage-engine:5.5.2")
+
+    testImplementation("org.hamcrest:hamcrest:2.2")
+}
+
+sourceSets {
+    create("functionalTest") {
+        java.srcDir(file("src/functionalTest/java"))
+        compileClasspath += sourceSets["main"].output
+        runtimeClasspath += sourceSets["main"].output
+    }
+    create("crossScalaVersionTest") {
+        java.srcDir(file("src/crossScalaVersionTest/java"))
+        compileClasspath += sourceSets["main"].output + sourceSets["functionalTest"].output
+        runtimeClasspath += sourceSets["main"].output + sourceSets["functionalTest"].output
+    }
+}
+
+configurations {
+    named("functionalTestImplementation") {
+        extendsFrom(configurations["testImplementation"])
+    }
+    named("functionalTestRuntimeOnly") {
+        extendsFrom(configurations["testRuntimeOnly"])
+    }
+
+    named("crossScalaVersionTestImplementation") {
+        extendsFrom(configurations["testImplementation"])
+    }
+    named("crossScalaVersionTestRuntimeOnly") {
+        extendsFrom(configurations["testRuntimeOnly"])
+    }
+}
+
+val crossScalaVersionTest by tasks.registering(Test::class) {
+    description = "Runs the cross scala version functional test."
+    group = "verification"
+    testClassesDirs = sourceSets["crossScalaVersionTest"].output
+    classpath = sourceSets["crossScalaVersionTest"].runtimeClasspath
+    setForkEvery(1) // crucial to run every test in its own JVM
+
+    testLogging {
+        events("passed", "failed", "skipped")
+        showStandardStreams = System.getenv("CI") == "true"
+    }
+
+    mustRunAfter(tasks["test"])
+}
+tasks["check"].dependsOn(crossScalaVersionTest)
+
+val functionalTest by tasks.registering(Test::class) {
+    description = "Runs the functional tests."
+    group = "verification"
+    testClassesDirs = sourceSets["functionalTest"].output
+    classpath = sourceSets["functionalTest"].runtimeClasspath
+
+    testLogging {
+        events("passed", "failed", "skipped")
+        showStandardStreams = System.getenv("CI") == "true"
+    }
+
+    systemProperty("failOnWarning", project.hasProperty("failOnWarning"))
+
+    mustRunAfter(crossScalaVersionTest)
+}
+tasks["check"].dependsOn(functionalTest)
+
+gradlePlugin {
+    testSourceSets(sourceSets["functionalTest"], sourceSets["crossScalaVersionTest"])
+}
+
+val kotlindocJar by tasks.registering(Jar::class) {
+    from(tasks.dokkaHtml.get().outputDirectory)
```

Comment:
Note that instead of
```
from(tasks.dokkaHtml.get().outputDirectory)
dependsOn(tasks.dokkaHtml)
```
it is more idiomatic to rely on automatic dependencies propagation:
```
from(tasks.dokkaHtml)
```
Gradle will automatically add a dependency on the `dokkaHtml` task here if its provider is used directly as an input.